icicle/include/merkle-tree/merkle.cuh

```diff
@@ -0,0 +1,94 @@
+#pragma once
+#ifndef MERKLE_H
+#define MERKLE_H
+
+#include "gpu-utils/device_context.cuh"
+#include "gpu-utils/error_handler.cuh"
+#include "utils/utils.h"
+#include "hash/hash.cuh"
+
+#include <iostream>
+#include <math.h>
+
+using namespace hash;
```

Comment:
I would avoid it since it makes the code harder to read and also for every file that includes it. Up to you. I generally prefer avoiding 'using namespace' especially in headers.